crates/context/config/src/client/starknet.rs

```diff
@@ -165,4 +190,88 @@ impl Network {
             }),
         }
     }
+
+    async fn mutate(&self, contract_id: &str, method: &str, args: Vec<u8>) {
+        let sender_address: Felt = Felt::from_str(&self.account_id)
+            .unwrap_or_else(|_| panic!("Failed to convert sender address to felt type"));
+        let secret_key: Felt = Felt::from_str(self.secret_key.as_str())
+            .unwrap_or_else(|_| panic!("Failed to convert sender address to felt type"));
+
+        let nonce = self
+            .get_nonce(self.account_id.as_str())
+            .await
+            .unwrap_or_else(|_| panic!("Failed to get nonce"));
+
+        let contract_id = Felt::from_str(contract_id)
+            .unwrap_or_else(|_| panic!("Failed to convert contract id to felt type"));
+
+        let entry_point_selector = get_selector_from_name(method)
+            .unwrap_or_else(|_| panic!("Failed to convert method name to entry point selector"));
+
+        let calldata: Vec<Felt> = if args.is_empty() {
+            vec![]
+        } else {
+            args.chunks(32)
+                .map(|chunk| {
+                    let mut padded_chunk = [0u8; 32];
+                    for (i, byte) in chunk.iter().enumerate() {
+                        padded_chunk[i] = *byte;
+                    }
+                    Felt::from_bytes_be(&padded_chunk)
+                })
+                .collect()
+        };
+
+        let transaction_hash = compute_transaction_hash(
+            sender_address,
+            contract_id,
+            entry_point_selector,
+            calldata.clone(),
```

Comment:
you can save this extra allocation by: ..._

```suggestion
            &calldata,
```

Reply:
updated